packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -83,16 +83,16 @@ const TooltipProvider: React.FC<TooltipProviderProps> = (
   return (
     <TooltipProviderContextProvider
       scope={__scopeTooltip}
-      isOpenDelayed={isOpenDelayed}
+      isOpenDelayedRef={isOpenDelayedRef}
       delayDuration={delayDuration}
       onOpen={React.useCallback(() => {
         window.clearTimeout(skipDelayTimerRef.current);
-        setIsOpenDelayed(false);
+        isOpenDelayedRef.current = false;
       }, [])}
       onClose={React.useCallback(() => {
         window.clearTimeout(skipDelayTimerRef.current);
         skipDelayTimerRef.current = window.setTimeout(
-          () => setIsOpenDelayed(true),
+          () => isOpenDelayedRef.current = true,
           skipDelayDuration
         );
```

Comment:
it's not the purpose of this PR so I'd rather keep it untouched.

Also they aren't created on every render. They are created only when `onOpen`/`onClose` gets called

Reply:
ah, or you suggest me to use `useCallbackRef` for `onOpen`/`onClose` instead of regular `useCallback`?

that makes more sense but still I think it's not so related to the scope of this PR. The less changes there are - the more chances it gets merged

Reply:
No, I meant extracting the arrow function into a self-declared function so that it’s not recreated on every render. [REF](https://legacy.reactjs.org/docs/faq-functions.html#:~:text=Is%20it%20OK%20to%20use,%2C%20by%20all%20means%2C%20optimize!)

I don't think this will be an issue at an atomic level, but in a wider context/large project, any sort of optimisation is welcomed. Tehn again, a separate PR can be created to avoid polluting this one

Reply:
Sorry, I think I am confusing you. With create a reference of the methods i meant something like this:

```ts
const TooltipProvider: React.FC<TooltipProviderProps> = (
  props: ScopedProps<TooltipProviderProps>
) => {
  const onOpen = React.useCallback(() => {
    window.clearTimeout(skipDelayTimerRef.current);
    isOpenDelayedRef.current = false;
  }, []);

  const onClose = React.useCallback(() => {
    window.clearTimeout(skipDelayTimerRef.current);
    skipDelayTimerRef.current = window.setTimeout(
      () => isOpenDelayedRef.current = true,
      skipDelayDuration
    );
  }, [skipDelayDuration]);

  ...others

return (
    <TooltipProviderContextProvider
      scope={__scopeTooltip}
      isOpenDelayedRef={isOpenDelayedRef}
      delayDuration={delayDuration}
      onOpen={onOpen}
      onClose={onClose}
```
it's same thing, we are just moving things around.

As for the docs, I shared the deprecated ones as the new one does not seem to have this but the concept is the same, class/funcion based component as well as vanilla js. 

For smaller projects this has no effect but if you are working on a very large project, chances you are, you would want to optimise this. Again, one for future improvements.

Reply:
as you said, it's same thing) 

```
const Component () => {
  const cb = useCallback(() => {}, [])
  
  return <ComponentB  onEvent={cb}/>
}
```
is identical to
```
const Component () => {
  return <ComponentB  onEvent={useCallback(() => {}, [])}/>
}
```
The function defined in `useCallback` will be created on every render in both cases.


If we are speaking about micro optimisations the latter can be even more optimised since we don't create extra variable for storing callback